### PR TITLE
Refactor Gene Annotation Project: Modularization and Enhanced Flexibility for Multiple Databases.

### PR DESCRIPTION
**Pull request by eyasubirhanu**

### CODE REVIEW

File: app/services/schema_data.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import yaml
+
+# Setup basic logging
+logging.basicConfig(level=logging.DEBUG)
+
+class SchemaManager:
+    def __init__(self, schema_config_path: str, biocypher_config_path: str):
+        self.bcy = BioCypher(schema_config_path=schema_config_path, biocypher_config_path=biocypher_config_path)
+        self.schema = self.bcy._get_ontology_mapping()._extend_schema()
+        self.parent_nodes = ["position entity", "coding element", "non coding element", "genomic variant", "epigenomic feature", "3d genome structure", "ontology term", "chromosome chain"]
```
Comment (tanksha) [inline app/services/schema_data.py:12]: consider automating the parent nodes and edges list generation from the schema. Maybe in another PR

**Review by tanksha — COMMENTED** [review]
A quick review, this looks good, a few minor comments about the newline at the end of files. Additionally, we can have the datasets available from the Bizon server for local testing

**Comment by Habush** [comment]
Hi Eyasu, it's good to see that you refactored the code. However, I don't see any implementation of the support for multiple databases in this PR as mentioned in its title. The `MeTTa_Query_Generator` class should inherit from an interface that specifies a generic way of generating queries, but that's not included in this PR. Are you planning this for another PR?

**Comment by eyasubirhanu** [comment]
Hey @abdu, Thank you for the feedback. The current implementation is
designed to support multiple databases through its modular structure and
flexibility. While the Interface isn't explicitly included in this PR, the
code is already set up for easy integration with different databases. I'll
include the interface in a follow-up PR. Additionally, I'm currently
working on integrating Prolog database support. What are your thoughts on
this?

**Pull request merged**